eth2/utils/bls/src/signature_bytes.rs

```diff
@@ -0,0 +1,106 @@
+use std::cmp::min;
+use std::fmt;
+
+use serde::de::{Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+
+use serde_hex::HexVisitor;
+use ssz::{Decode, DecodeError, Encode, ssz_encode};
+
+use super::{BLS_SIG_BYTE_SIZE, Signature};
+
+#[derive(Clone)]
+pub struct SignatureBytes([u8; BLS_SIG_BYTE_SIZE]);
+
+impl SignatureBytes {
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        Ok(Self::new_from_bytes(bytes))
+    }
+
+    pub fn new_from_bytes(bytes: &[u8]) -> Self {
+        Self(Self::get_bytes(bytes))
+    }
+
+    pub fn empty_signature() -> Self {
+        Self::new_from_bytes(&[])
+    }
+
+    pub fn new(signature: Signature) -> Self {
+        Self(Self::get_bytes(signature.as_bytes().as_slice()))
+    }
+
+    pub fn as_bytes(&self) -> Vec<u8> {
+        self.0.to_vec()
+    }
+
+    pub fn parse_signature(&self) -> Result<Signature, DecodeError> {
```

Comment:
I'd be tempted to use a [`TryInto`](https://doc.rust-lang.org/std/convert/trait.TryInto.html) impl instead of this function, primarily because it will allow for easier compatibility with the `fake_crypto` feature.

If we do `TryInto`/`TryFrom` for `Signature` and `FakeSignature` it'll avoid having to create a `parse_signature -> Signature` and `parse_signature -> FakeSignature` (with a `cfg` attribute doing the switch for `fake_crypto`).

Reply:
I implemented now TryInto<Signature> for &SignatureBytes and From<Signature> for SignatureBytes. Or do I need to add anything else? I am not very familiar with the fake_crypto concept.